test/Microsoft.AspNetCore.SignalR.Tests/TestEndPoint.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Sockets;
+
+namespace Microsoft.AspNetCore.SignalR.Test.Server
+{
+    public class TestEndPoint : EndPoint
+    {
+        public ConnectionList Connections { get; } = new ConnectionList();
+
+        public override async Task OnConnectedAsync(ConnectionContext connection)
+        {
+            while (await connection.Transport.In.WaitToReadAsync())
+            {
+                if (connection.Transport.In.TryRead(out var buffer))
```

Comment:
Make this a `while` loop. Not a big deal here but it means you don't have to async wait if multiple messages are available at once. Doesn't really matter for this test code, but it's a good pattern to have :)